src/Microsoft.AspNetCore.SignalR.Redis/Internal/RedisProtocol.cs

```diff
@@ -0,0 +1,166 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using Microsoft.AspNetCore.SignalR.Internal;
+using Microsoft.AspNetCore.SignalR.Internal.Protocol;
+
+namespace Microsoft.AspNetCore.SignalR.Redis.Internal
+{
+    public static class RedisProtocol
+    {
+        private static readonly Encoding _utf8NoBom = new UTF8Encoding(encoderShouldEmitUTF8Identifier: false);
+
+        // REVIEW: This protocol is simplistic and probably inefficient as is. Discuss.
```

Comment:
I'm ok with that as long as we have benchmarks that let us measure improvements.